specutils/tests/test_loaders.py

```diff
@@ -1273,3 +1273,85 @@ def test_galah_guess(remote_data_path):
         assert isinstance(spectra[1].uncertainty, StdDevUncertainty)
         assert spectra[1].meta.get("label") is not None
         assert spectra[1].meta.get("header") is not None
+
+
+# We cannot use remote_access directly in the MIRI MRS tests, because
+# the test functions are called once for every file in the remote access
+# list, but we need to feed the entire set to the functions at once. We
+# store the file names in a list and use a dummy test method to load the
+# list via the remote_access machinery.
+#
+# MIRI MRS 1D data sets are comprised of 12 files.
+# We add one bad file to test the skip/warn on missing file functionality
+filename_list = ["bad_file.fits"]
+@remote_access([
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch1-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch1-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch1-short_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch2-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch2-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch2-short_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch3-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch3-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch3-short_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch4-long_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch4-medium_x1d.fits'},
+    {'id': '5082863', 'filename': 'combine_dithers_all_exposures_ch4-short_x1d.fits'},
+])
+def test_loaddata_miri_mrs(remote_data_path):
+    filename_list.append(remote_data_path)
+
+
+# loading from a list of file names
+@pytest.mark.remote_data
+def test_spectrum_list_names_miri_mrs(caplog):
+
+    # Format is explicitly set
+    with pytest.raises(FileNotFoundError):
+        specs = SpectrumList.read(filename_list, format="JWST x1d MIRI MRS")
+
+    # Skip missing file silently
+    specs = SpectrumList.read(filename_list, format="JWST x1d MIRI MRS", missing='silent')
+
+    assert len(specs) == 12
+    for spec in specs:
+        assert isinstance(spec, Spectrum1D)
+        assert spec.spectral_axis.unit == u.Unit("um")
```

Comment:
No need for overhead to pass string into constructor.
```suggestion
        assert spec.spectral_axis.unit == u.micron
```